src/server/handlers/launchpad.js

```diff
@@ -545,9 +561,46 @@ export const findSnaps = async (req, res) => {
       );
       return { ...snap, snapcraft_data: snapcraftData };
     }));
-    if (req.session.user) {
-      await initializeMetrics(req.session.user.id, snaps);
-    }
+    await db.transaction(async (trx) => {
```

Comment:
Thanks.  I've pushed a change to refactor this into what should be a more readable form.